src/js/modules/accounts/actions.js

```diff
@@ -0,0 +1,123 @@
+import axios from "axios";
+import appHistory from "../../tools/appHistory";
+
+import * as t from "./actionTypes";
+import { STUY_SPEC_API_URL, STUY_SPEC_API_HEADERS } from "../../constants";
+
+export const signUp = values => {
+  return dispatch => {
+    dispatch({
+      type: t.SIGN_UP_PENDING,
+      payload: values,
+    });
+    axios
+      .post(`${STUY_SPEC_API_URL}/auth`, values, STUY_SPEC_API_HEADERS)
+      .then(response => {
+        dispatch({
+          type: t.SIGN_UP_FULFILLED,
+          payload: response,
+        });
+        const user = response.data.data;
+        /* As Devise only accepts email/passwords, we need a separate update
+         * for other user properties. Note that we are not simply dispatching
+         * an UPDATE_USER because that will change the status's current form
+         * name to editUser, which will prevent success text from rendering
+         * on the signUpForm.
+         */
+        axios.put(
```

Comment:
You're posting, then putting the same exact values. POST the email and password, then PUT the other info. Let's limit the number of requests with sensitive data. Also, consider making a separate action then calling it here